homeassistant/components/alarm_control_panel/manual.py

```diff
@@ -24,9 +25,28 @@
 DEFAULT_TRIGGER_TIME = 120
 DEFAULT_DISARM_AFTER_TRIGGER = False
 
+SUPPORTED_PENDING_STATES = [STATE_ALARM_ARMED_AWAY, STATE_ALARM_ARMED_HOME,
+                            STATE_ALARM_ARMED_NIGHT, STATE_ALARM_TRIGGERED]
+
 ATTR_POST_PENDING_STATE = 'post_pending_state'
 
-PLATFORM_SCHEMA = vol.Schema({
+
+def _state_validator(config):
+    for state in SUPPORTED_PENDING_STATES:
+        if CONF_PENDING_TIME not in config[state]:
+            config = copy.deepcopy(config)
```

Comment:
Why is the copy needed?

Reply:
To be honest I have no idea why it's needed. Initially I don't have it but when I ran my unit tests, almost all of them failed and it likes the config are polluted from previous tests e.g. it's getting values from previous tests.

I see the example you provided (https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/scene/__init__.py#L26-L53) is returning a new copy of config when it needs to modify so I thought maybe that's what's needed and it worked for me.

Do you know what might cause this?

Reply:
I haven't figured it out, but I have tested it. You should be able to move the line to the beginning of this validator before starting the iteration of states in `SUPPORTED_PENDING_STATES`. This will only copy the config one time.

Reply:
Ok makes sense, I just moved it out of the loop now